standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -1497,8 +1497,8 @@ GetPartitionsPsWithAuthResponse listPartitionsWithAuthInfoRequest(GetPartitionsP
    * @throws MetaException error accessing the RDBMS.
    * @throws TException thrift transport error
    */
-  List<Partition> getPartitionsByNames(String db_name, String tbl_name,
-      List<String> part_names) throws NoSuchObjectException, MetaException, TException;
+  List<Partition> getPartitionsByNames(String db_name, String tbl_name, List<String> part_names,
```

Comment:
Just change an interface may cause a backward compatible issue, maybe just add a new one and keeps old ones?

Reply:
No, we want everyone to send validWriteIdList and tableId going forward. So, creating new interfaces doesn't work.

Reply:
How about old version clients?

Reply:
I am not changing HMS APIs here.

Reply:
@yongzhi Please check the latest. I have retained all the existing IMetaStoreClient APIs and introduced new ones. Also, got a green build.